### ISSUE TITLE
Add support for SSH commit signing

### ISSUE THREAD

**Issue by drewdeponte** [uptech/git-ps-rs#45]
Add the ability to sign commits created by git-ps-rs using SSH keys. This is similar to signing with GPG but just with SSH keys instead.

**Comment by Alizter** [comment]
Any workaround in the meantime?

**Comment by drewdeponte** [comment]
`gps` is GPG aware and when it rebases it does GPG sign the commits if you have it enabled.

Can you verify locally that the commits are signed?

**Comment by Alizter** [comment]
I have this set:
```
git config --global commit.gpgsign true
```

and I am able to commit and push to Github myself no problem. It shows up with the verified mark. When I do `gps rr` however the commits are not signed. Could this be something to do with the GitHub CLI?

**Comment by Alizter** [comment]
Ah I see the problem. I didn't have that option signed actually, I was apparently using a git alias!

**Comment by drewdeponte** [comment]
Ok cool that is telling you it is failing to get the user signing key from the git config.

That error comes from this line of code, https://github.com/uptech/git-ps-rs/blob/main/src/ps/private/git.rs#L202

It is saying that it is failing to get the `user.signingkey` config value.

In the `~/.gitconfig` it should look like the following.

```
[user]
    signingkey = YOUR-ACTUAL-GPG-SIGNING-KEY-ID-HERE
```

**Comment by Alizter** [comment]
Yes! Just noticed that I didn't have that too. I've added it in and it is working now.

**Comment by drewdeponte** [comment]
Lol, no problem. Glad it is working now.

**Comment by ctsstc** [comment]
With 1password launching their Github integration with SSH signed commits, there may be more pressure for this:

> 1pass + GH integration for SSH Signed Commits :tada:  big hype :rocket:
> https://www.youtube.com/watch?v=BMFvhl0WRFQ
